internal/scheduled/runner.go

```diff
@@ -70,26 +76,67 @@ func NewRunner(logger Logger, store ScheduledStore, donorStore DonorStore, notif
 	return r
 }
 
-func (r *Runner) Run(ctx context.Context) error {
+func (r *Runner) Run(ctx context.Context) (cloudwatch.PutMetricDataInput, error) {
 	r.waiter.Reset()
 
+	processedTasks := float64(0)
+	ignoredTasks := float64(0)
+	errorTasks := float64(0)
+	start := r.now()
+
 	for {
 		row, err := r.store.Pop(ctx, r.now())
 
 		if errors.Is(err, dynamo.NotFoundError{}) {
 			r.logger.InfoContext(ctx, "no scheduled tasks to process")
-			return nil
+
+			var metrics cloudwatch.PutMetricDataInput
+
+			if processedTasks > 0 || errorTasks > 0 || ignoredTasks > 0 {
+				elapsed := r.since(start)
+
+				metrics = cloudwatch.PutMetricDataInput{
```

Comment:
it might be nicer to abstract this a little bit, you could have `(*Runner).Processed/Ignored/Errored()` methods to increment the counters and a `(*Runner).Metrics()` to return the `cloudwatch.PutMetricDataInput`, or something like that